python/CSXCAD/CSProperties.pyx

```diff
@@ -388,16 +389,30 @@ cdef class CSProperties:
         """
         return self.__CreatePrimitive(c_CSPrimitives.POLYHEDRON, **kw)
 
-    def AddPolyhedronReader(self, filename, **kw):
-        """ AddPolyhedronReader(filename, **kw)
+    def AddPolyhedronReader(self, file:Path, **kw)->CSPrimitives:
+        """Add a polyhedron from file to this property.
 
-        Add a polyhedron from file to this property.
+        Arguments
+        ---------
+        file:
+            The path to the file to add.
+
+        Returns
+        -------
+        primitive:
+            An object of type `CSPrimitives` with the primitive just created.
 
         See Also
         --------
         CSXCAD.CSPrimitives.CSPrimPolyhedronReader : See here for details on primitive arguments
         """
-        return self.__CreatePrimitive(c_CSPrimitives.POLYHEDRONREADER, filename=filename, **kw)
+        file = Path(file) # Make sure that whatever we receive can be interpreted as a path to a file.
+        if not file.is_file():
+            raise FileNotFoundError(f'`file` points to a file that does not exist. Received {repr(str(file))}, absolute path is {repr(str(file.resolve()))}. ')
```

Comment:
Does it actually matter here if the file exists or not? For example we just define the file name, it is then later created and then maybe read? Why check the existence here if it does not need to exist (yet)? 
Again this feels unnecessary restrictive?

Reply:
I think that a [fail-fast approach](https://en.wikipedia.org/wiki/Fail-fast_system) makes it easy for the user. Are there many cases in which the file would be created after the configuration is done? Not an expert here, but it sounds to me that the answer is no.